### PR TITLE
40008

### PR DESCRIPTION
**Pull request by saimekala07**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [40008](https://devops.dchbx.org/redmine/issues/40008) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | n/a |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | Philip Burrowes |

### CODE REVIEW

File: app/models/family.rb [hunk 1]
```diff
@@ -293,6 +293,11 @@ def primary_family_member=(new_primary_family_member)
     primary_family_member
   end
 
+  def terminated_enrollments
+    return [] if  latest_household.blank?
+    enrollments.order_by(:created_at => "DESC").select{|hbx| ["coverage_terminated", "coverage_termination_pending"].include?(hbx.aasm_state)}
```
Comment (raghuramg) [inline app/models/family.rb]: Use can_update_enrollments_end_date scope

File: app/views/exchanges/hbx_profiles/_update_enrollment_termianted_on_date.html.erb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<table class="table table-striped">
+  <%if @result[:success].present?%>
```
Comment (raghuramg) [inline app/views/exchanges/hbx_profiles/_update_enrollment_termianted_on_date.html.erb]: @result always have single enrollment. Remove the loop here and also remove duplication of code inside this template.

File: app/views/exchanges/hbx_profiles/_view_enrollment_to_update_end_date.html.erb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        jQuery(jQuery(this).closest('tr').find('[type=checkbox]')[0]).prop('checked', false);
+        jQuery(jQuery(this).closest('tr').find('[type=text]')[0]).prop('disabled', true);
+      }
+    });
+  });
+
+</script>
+
+<td colspan="90%">
+  <%if @enrollments .present?%>
```
Comment (raghuramg) [inline app/views/exchanges/hbx_profiles/_view_enrollment_to_update_end_date.html.erb]: syntax needs correction

File: app/views/exchanges/hbx_profiles/_update_enrollment_termianted_on_date.html.erb (hunk 2 above)
Comment (saimekala07) [reply app/views/exchanges/hbx_profiles/_update_enrollment_termianted_on_date.html.erb]: updated as per suggestion.

File: app/models/family.rb (hunk 1 above)
Comment (saimekala07) [reply app/models/family.rb]: updated method to use scope.

File: app/views/exchanges/hbx_profiles/_view_enrollment_to_update_end_date.html.erb (hunk 3 above)
Comment (saimekala07) [reply app/views/exchanges/hbx_profiles/_view_enrollment_to_update_end_date.html.erb]: fixed typo

File: app/controllers/exchanges/hbx_profiles_controller.rb [hunk 4]
```diff
@@ -334,6 +334,38 @@ def update_terminate_enrollment
...
+          enrollment.terminate_coverage!(termination_date)
+          enrollment.notify_enrollment_cancel_or_termination_event(params["edi_required"].present?)
+          @result[:success] << enrollment
+        else
+          @result[:failure] << enrollment
+        end
+      end
+    rescue Exception => e
+      redirect_to exchanges_hbx_profiles_root_path, flash: {error: e.to_s}
+      return
```
Comment (raghuramg) [inline app/controllers/exchanges/hbx_profiles_controller.rb]: can you remove this return statement

File: app/controllers/exchanges/hbx_profiles_controller.rb [hunk 5]
```diff
@@ -334,6 +334,38 @@ def update_terminate_enrollment
...
+    @enrollments = @person.primary_family.terminated_enrollments
+  end
+
+  def update_enrollment_termianted_on_date
+    begin
+      enrollment = HbxEnrollment.find(params[:enrollment_id].strip)
+      @row = params[:family_actions_id]
+      @result = {success: [], failure: []}
+      termination_date = Date.strptime(params["new_termination_date"], "%m/%d/%Y")
+      if enrollment.present? && (enrollment.coverage_terminated? || enrollment.coverage_termination_pending?)
```
Comment (raghuramg) [inline app/controllers/exchanges/hbx_profiles_controller.rb]: move this logic into enrollment model...and have it return hash as response....with success/failure

**Comment by harshared** [comment]
closing since code is already merged in endate-reinstate-rc

**Pull request closed without merging**